File: parse_pdfs.ts
import { LlamaParseReader } from "llamaindex";
import * as fs from 'fs/promises';
import { resolve, parse } from 'path';

const API_KEY = "llx-4xn9ODNoVjMWWu3MsyDfs0NxiPknULsr5TtAklFx79YeGgEb";
process.env.LLAMA_CLOUD_API_KEY = API_KEY;

async function main() {
    const inputDir = 'D:/Projects/LinhIQ/apps/data/curriculum/igcse/biology/pdf';
    const outputDir = 'D:/Projects/LinhIQ/apps/data/curriculum/igcse/biology/markdown';
    
    // Create outputDir if it doesn't exist
    await fs.mkdir(outputDir, { recursive: true });

    const files = await fs.readdir(inputDir);
    const pdfFiles = files.filter(f => f.toLowerCase().endsWith('.pdf'));

    const reader = new LlamaParseReader({ resultType: "markdown" });

    for (const file of pdfFiles) {
        console.log(`Processing ${file}...`);
        const inputPath = resolve(inputDir, file);
        const outputPath = resolve(outputDir, parse(file).name + '.md');
        
        try {
            const documents = await reader.loadData(inputPath);
            const markdownContent = documents.map(d => d.text).join('\n\n');
            await fs.writeFile(outputPath, markdownContent, 'utf-8');
            console.log(`Saved markdown to ${outputPath}`);
        } catch (error) {
            console.error(`Error processing ${file}:`, error);
        }
    }
}

main().catch(console.error);

File: parse_pdfs.py
import os
import glob
from llama_parse import LlamaParse

os.environ["LLAMA_CLOUD_API_KEY"] = "llx-4xn9ODNoVjMWWu3MsyDfs0NxiPknULsr5TtAklFx79YeGgEb"

def main():
    input_dir = r"D:\Projects\Javirs\apps\data\curriculum\igcse\biology\pdf"
    output_dir = r"D:\Projects\Javirs\apps\data\curriculum\igcse\biology\textbook"

    if not os.path.exists(output_dir):
        os.makedirs(output_dir)

    pdf_files = glob.glob(os.path.join(input_dir, "*.pdf"))
    
    if not pdf_files:
        print(f"No PDF files found in {input_dir}")
        return

    parser = LlamaParse(
        result_type="markdown",
        verbose=True
    )

    for pdf_file in pdf_files:
        print(f"Parsing {pdf_file} ...")
        filename = os.path.basename(pdf_file)
        name, _ = os.path.splitext(filename)
        output_file = os.path.join(output_dir, f"{name}.md")
        
        try:
            documents = parser.load_data(pdf_file)
            markdown_content = "\n\n".join([doc.text for doc in documents])
            
            with open(output_file, "w", encoding="utf-8") as f:
                f.write(markdown_content)
                
            print(f"Successfully saved markdown to {output_file}")
        except Exception as e:
            print(f"Error parsing {pdf_file}: {e}")

if __name__ == "__main__":
    main()
